Argocd rollback guide (#941)
port-labs/port-docs: docs/create-self-service-experiences/setup-backend/github-workflow/examples/rollback-argocd-deployment.md

---
sidebar_position: 5
---

import PortTooltip from "/src/components/tooltip/tooltip.jsx";

# Rollback ArgoCD Deployment

> In the following guide, you are going to create a self-service action in Port that executes a GitHub workflow to rollback an ArgoCD deployment. The rollback involves updating the deployment manifest with a new container image and creating a Github pull request (PR) for it. The workflow can optionally merge the PR when enabled.

Reviewer: ```suggestion
In this guide, we will create a self-service action in Port that executes a GitHub workflow to rollback an ArgoCD deployment. The rollback involves updating the deployment manifest with a new container image and creating a Github pull request (PR) for it. The workflow can optionally merge the PR when enabled.
```

## Prerequisites
1. Install Port's GitHub app by clicking [here](https://github.com/apps/getport-io/installations/new)

> 2. This guide assumes the presence of a `Service` and `Image` blueprint representing your repository where ArgoCD lives and your container image

Reviewer: How should the user have these blueprints? Please add a link to the relevant guide/page

Author: This is already included in the doc:

<img width="873" alt="Screenshot 2024-03-04 at 11 43 28 AM" src="https://github.com/port-labs/port-docs/assets/15999660/ae67a738-266f-4301-a9eb-b8340e886048">
